File: flutter/lib/screens/home_screen.dart
import 'package:flutter/material.dart';
import '../state/catalog_state.dart';

class HomeScreen extends StatelessWidget {
  final CatalogState state;
  final Function(int) onNavigate;

  const HomeScreen({
    super.key,
    required this.state,
    required this.onNavigate,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return ListenableBuilder(
      listenable: state,
      builder: (context, _) {
        return SingleChildScrollView(
          padding: const EdgeInsets.all(16.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              // Banner institucional
              Card(
                color: theme.colorScheme.primaryContainer,
                shape: RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(16),
                ),
                child: Padding(
                  padding: const EdgeInsets.all(20.0),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        'ESCOM - IPN | DAMN',
                        style: theme.textTheme.labelLarge?.copyWith(
                          color: theme.colorScheme.onPrimaryContainer,
                        ),
                      ),
                      const SizedBox(height: 4),
                      Text(
                        'Catálogo UI en Flutter (Dart)',
                        style: theme.textTheme.headlineSmall?.copyWith(
                          fontWeight: FontWeight.bold,
                          color: theme.colorScheme.onPrimaryContainer,
                        ),
                      ),
                      const SizedBox(height: 8),
                      Text(
                        'Construcción de interfaces móviles con árbol reactivo de widgets (Material 3). Cubre más de 35 elementos interactivos en seis categorías, adaptación a modo claro/oscuro y reactividad en tiempo real.',
                        style: theme.textTheme.bodyMedium?.copyWith(
                          color: theme.colorScheme.onPrimaryContainer,
                        ),
                      ),
                      const SizedBox(height: 12),
                      Text(
                        'Alumno: Aragón Martínez Manuel Alejandro | Boleta: 2023630411 | Grupo: 7CV4',
                        style: theme.textTheme.labelSmall?.copyWith(
                          color: theme.colorScheme.onPrimaryContainer,
                        ),
                      ),
                    ],
                  ),
                ),
              ),

              const SizedBox(height: 16),

              // Tarjeta de estado reactivo global
              Card(
                elevation: 0,
                shape: RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(12),
                  side: BorderSide(color: theme.colorScheme.outlineVariant),
                ),
                child: Padding(
                  padding: const EdgeInsets.all(16.0),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        'Estado Global de la Aplicación',
                        style: theme.textTheme.titleSmall?.copyWith(
                          color: theme.colorScheme.primary,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                      const SizedBox(height: 4),
                      Text(
                        'Última acción: ${state.lastAction}',
                        style: theme.textTheme.bodyMedium,
                      ),
                      const SizedBox(height: 4),
                      Text(
                        'Total de elementos en la colección: ${state.items.length}',
                        style: theme.textTheme.labelSmall?.copyWith(
                          color: theme.colorScheme.secondary,
                        ),
                      ),
                    ],
                  ),
                ),
              ),

              const SizedBox(height: 20),

              Text(
                'Secciones del Catálogo',
                style: theme.textTheme.titleMedium,
              ),

              const SizedBox(height: 12),

              _buildNavButton(
                context: context,
                index: 1,
                title: '1. Entrada de Texto (Campos, Validación, Teclados)',
                icon: Icons.text_fields,
              ),
              _buildNavButton(
                context: context,
                index: 2,
                title: '2. Botones y Acciones (Filled, Outlined, FAB, Loading)',
                icon: Icons.smart_button,
              ),
              _buildNavButton(
                context: context,
                index: 3,
                title: '3. Elementos de Selección (Check, Radio, Sliders, Pickers)',
                icon: Icons.radio_button_checked,
              ),
              _buildNavButton(
                context: context,
                index: 4,
                title: '4. Listas y Colecciones (ListView, Grid, Swipe, Tabs)',
                icon: Icons.list,
              ),
              _buildNavButton(
                context: context,
                index: 5,
                title: '5. Información y Retroalimentación (Progress, Modales)',
                icon: Icons.info,
              ),
              _buildNavButton(
                context: context,
                index: 6,
                title: '6. Contenedores y Estructura (Row, Column, Stack, Flex)',
                icon: Icons.grid_view,
              ),
            ],
          ),
        );
      },
    );
  }

  Widget _buildNavButton({
    required BuildContext context,
    required int index,
    required String title,
    required IconData icon,
  }) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 4.0),
      child: FilledButton.tonal(
        onPressed: () => onNavigate(index),
        style: FilledButton.styleFrom(
          alignment: Alignment.centerLeft,
          padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 14),
          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
        ),
        child: Row(
          children: [
            Icon(icon),
            const SizedBox(width: 12),
            Expanded(
              child: Text(
                title,
                style: const TextStyle(fontWeight: FontWeight.w500),
              ),
            ),
            const Icon(Icons.arrow_forward_ios, size: 14),
          ],
        ),
      ),
    );
  }
}

File: android-compose/app/src/main/java/com/escom/tarea2/compose/ui/screens/HomeScreen.kt
package com.escom.tarea2.compose.ui.screens

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.GridView
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.List
import androidx.compose.material.icons.filled.RadioButtonChecked
import androidx.compose.material.icons.filled.SmartButton
import androidx.compose.material.icons.filled.TextFields
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedCard
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.escom.tarea2.compose.viewmodel.SharedCatalogViewModel

@Composable
fun HomeScreen(
    viewModel: SharedCatalogViewModel,
    onNavigateToSection: (String) -> Unit
) {
    val lastAction by viewModel.lastAction.collectAsState()
    val items by viewModel.items.collectAsState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(16.dp)
    ) {
        // Tarjeta institucional
        Card(
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.primaryContainer,
                contentColor = MaterialTheme.colorScheme.onPrimaryContainer
            )
        ) {
            Column(modifier = Modifier.padding(20.dp)) {
                Text(
                    text = "ESCOM - IPN | DAMN",
                    style = MaterialTheme.typography.labelLarge
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = "Catálogo UI en Jetpack Compose",
                    style = MaterialTheme.typography.headlineMedium
                )
                Spacer(modifier = Modifier.height(8.dp))
                Text(
                    text = "Implementación declarativa y reactiva en Kotlin con Jetpack Compose y Material 3. Demostración interactiva de más de 35 elementos organizados en seis secciones.",
                    style = MaterialTheme.typography.bodyMedium
                )
                Spacer(modifier = Modifier.height(12.dp))
                Text(
                    text = "Alumno: Aragón Martínez Manuel Alejandro | Boleta: 2023630411 | Grupo: 7CV4",
                    style = MaterialTheme.typography.labelSmall
                )
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        // Tarjeta de estado global reactivo
        OutlinedCard(
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(12.dp)
        ) {
            Column(modifier = Modifier.padding(16.dp)) {
                Text(
                    text = "Estado Global de la Aplicación",
                    style = MaterialTheme.typography.titleMedium,
                    color = MaterialTheme.colorScheme.primary
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = "Última acción: $lastAction",
                    style = MaterialTheme.typography.bodyMedium
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = "Total de registros en colección: ${items.size}",
                    style = MaterialTheme.typography.labelSmall,
                    color = MaterialTheme.colorScheme.secondary
                )
            }
        }

        Spacer(modifier = Modifier.height(20.dp))

        Text(
            text = "Explorar Categorías",
            style = MaterialTheme.typography.titleMedium
        )

        Spacer(modifier = Modifier.height(12.dp))

        val sections = listOf(
            Triple("1. Entrada de Texto", "text_inputs", Icons.Default.TextFields),
            Triple("2. Botones y Acciones", "buttons", Icons.Default.SmartButton),
            Triple("3. Elementos de Selección", "selection", Icons.Default.RadioButtonChecked),
            Triple("4. Listas y Colecciones", "lists", Icons.Default.List),
            Triple("5. Retroalimentación", "feedback", Icons.Default.Info),
            Triple("6. Contenedores y Estructura", "containers", Icons.Default.GridView)
        )

        sections.forEach { (title, route, icon) ->
            Button(
                onClick = { onNavigateToSection(route) },
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(vertical = 4.dp),
                shape = RoundedCornerShape(10.dp),
                colors = ButtonDefaults.filledTonalButtonColors()
            ) {
                Icon(imageVector = icon, contentDescription = null)
                Spacer(modifier = Modifier.weight(1f))
                Text(text = title)
            }
        }
    }
}
